### PR TITLE
feat(integrations): Add annotations for issue tracker integrations

### PR DESCRIPTION
**Pull request by MeredithAnya**
The new integrations didn't have annotations added so this PR adds them:

screen shot 2018-11-20 at 4 32 08 pm

cc @saragilford

### CODE REVIEW

File: src/sentry/integrations/issues.py [hunk 1]
```diff
@@ -235,6 +235,26 @@ def create_default_repo_choice(self, default_repo):
...
+
+        external_issues = ExternalIssue.objects.filter(
+            id__in=external_issue_ids,
+            integration_id=self.model.id,
+        )
+        annotations = []
+        for ei in external_issues:
+            link = self.get_issue_url(ei.key)
+            label = self.get_issue_display_name(ei) or ei.key
+            annotations.append('<a href="%s">%s</a>' % (link, label))
```
Comment (markstory) [inline src/sentry/integrations/issues.py]: Should we consider not shipping HTML to the client an send a dictionary of `url` and `label` that we can turn into HTML?

File: src/sentry/integrations/issues.py [hunk 2]
```diff
@@ -235,6 +235,26 @@ def create_default_repo_choice(self, default_repo):
...
+            group_id=group.id,
+            project_id=group.project_id,
+            linked_type=GroupLink.LinkedType.issue,
+            relationship=GroupLink.Relationship.references,
+        ).values_list('linked_id', flat=True)
+
+        external_issues = ExternalIssue.objects.filter(
+            id__in=external_issue_ids,
+            integration_id=self.model.id,
+        )
```
Comment (markstory) [inline src/sentry/integrations/issues.py]: I'm guessing django doesn't give us a reasonable way to query both these tables with one query or a subquery?

File: src/sentry/api/serializers/models/group.py [hunk 3]
```diff
@@ -225,6 +226,17 @@ def get_attrs(self, item_list, user):
                     safe_execute(plugin.get_annotations, group=item, _with_transaction=False) or ()
                 )
 
+            for integration in Integration.objects.filter(
+                    organizations=item.project.organization_id):
+                if not (integration.has_feature(IntegrationFeatures.ISSUE_BASIC) or integration.has_feature(
+                        IntegrationFeatures.ISSUE_SYNC)):
+                    continue
```
Comment (markstory) [inline src/sentry/api/serializers/models/group.py]: Could this be `if hasattr('get_annotations', integration)`? Then implementing the method gets you the annotation and there are fewer stars to align when building new integrations.

File: src/sentry/integrations/issues.py (hunk 1 above)
Comment (MeredithAnya) [reply src/sentry/integrations/issues.py]: Maybe long term, but wanted to at least make it work with https://github.com/getsentry/sentry/blob/f1dc069482c36d738ac222ace4412274dd650fff/src/sentry/static/sentry/app/components/eventOrGroupExtraDetails.jsx#L115

File: src/sentry/api/serializers/models/group.py (hunk 3 above)
Comment (lauryndbrown) [reply src/sentry/api/serializers/models/group.py]: Hmm.. @markstory I think I support using the integration feature flags in this case, given its how we've handled other integration features. I like the idea from the standpoint of being able to search Issue_basic or Issue_sync to find all the places we've implemented those specific features as well from a practical maintainer's standpoint.

File: src/sentry/integrations/issues.py (hunk 2 above)
Comment (lauryndbrown) [reply src/sentry/integrations/issues.py]: Hmm.. outside of using `extra` I'm not sure there is a better way to do that. Though I would be careful with extra subqueries and mysql though... different versions don't support it if I recall correctly. I feel this is pretty clear and wouldn't advocate for going that route.

**Comment by lauryndbrown** [comment]
@MeredithAnya I'd love to see some tests for this change :).

File: src/sentry/api/serializers/models/group.py (hunk 3 above)
Comment (markstory) [reply src/sentry/api/serializers/models/group.py]: Ok.

**Review by markstory — APPROVED** [review]

**Pull request merged**